AutoRest/TestServer/swagger/custom-baseUrl-more-options.json

```diff
@@ -0,0 +1,109 @@
+﻿{
+  "swagger": "2.0",
+  "info": {
+    "title": "AutoRest Parameterized Custom Host Test Client",
+    "description": "Test Infrastructure for AutoRest",
+    "version": "1.0.0"
+  },
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{vault}{secret}{dnsSuffix}",
+    "useSchemePrefix": false,
+    "positionInOperation": "first",
+    "parameters": [
+      {
+        "name": "vault",
+        "description": "The vault name, e.g. https://myvault",
+        "required": true,
+        "type": "string",
+        "in": "path",
+        "x-ms-skip-url-encoding": true
+      },
+      {
+        "name": "secret",
+        "description": "Secret value.",
+        "required": true,
+        "type": "string",
+        "in": "path",
+        "x-ms-skip-url-encoding": true
+      },
+      {
+        "$ref": "#/parameters/dnsSuffix"
```

Comment:
@NiklasGustafsson - This is an example of a reference to global parameter that will be a property on the client.